src/pages/logos/LogoDeepSearch.jsx

```diff
@@ -90,16 +90,38 @@ export default function LogoSearch() {
   }, [searchState, searchCount]);
 
   React.useEffect(() => {
-    if (selectAll) {
+    if (selectedAll) {
       setLogosToAnnotate((prev) =>
-        prev.map((logo) => ({ ...logo, selected: true }))
+        // set current page to selected, keep old
+        prev.map((logo, index) => ({
+          ...logo,
+          selected:
+            index < page * pageSize && index >= (page - 1) * pageSize
+              ? true
+              : logo.selected,
+        }))
       );
     } else {
       setLogosToAnnotate((prev) =>
-        prev.map((logo) => ({ ...logo, selected: false }))
+        // set current page to unselected, keep old
+        prev.map((logo, index) => ({
+          ...logo,
+          selected:
+            index < page * pageSize && index >= (page - 1) * pageSize
+              ? false
+              : logo.selected,
+        }))
```

Comment:
I would keep the previous behavior for the unselect all.
```suggestion
        prev.map((logo) => ({ ...logo, selected: false }))
```

It's harder to correct an error than to add a new annotation. So I prefer that 
- **select all* is limited to what the user can see
- **unselect all** is extended to what users do not see and might have forget

But that's a personal opinion